eng/configure-toolset.ps1

```diff
@@ -0,0 +1,58 @@
+# If we are running a test pass, ensure that we point at the TestHost and that
+# tests are run against it.
+if ($test)
```

Comment:
where is test defined/set? what exactly is "test" (is it a bool?). I believe this will return true if test is a Boolean set to $false

Reply:
the same goes for all checks you have that are like this. i think there are better/safer powershell symantex for checking if a variable is defined

Reply:
Test is a switch defined in eng/common/build.ps1.  It's always true or false and works as expected.  You can tell because this never runs when the build is not explicitly called with -test.  In fact, this running during the build phase was what the original error that the yml change fixes.  So we've validated this inside and out at this point.

Reply:
I think adding a comment here to say where $test comes from is a good idea though.  Perhaps a better discussion of how this is an extension script from Arcade's build.ps1.

Reply:
Let me clarify above, its always $true or $false, so we're not going to run into the situation where if ($test) evaluates to true just because it is defined.

Reply:
In terms of the null checks, I should probably add `-ne $null` in those instances, I'll make that change.

Reply:
Made a few changes to params. comments, and usage throughout the script.  When there are vars that are null checked I made them explicit, otherwise I ensured that some vars are switches in order to more explicitly call out that they are always set.